### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by kopiczko**
Related discussions:

- Confirming this netpol is for cluster DNS in general not the k8s-dns-node-cache-app https://gigantic.slack.com/archives/C02GLN41UP2/p1711016179149589
- Confirming it works for cilium in ENI mode: https://gigantic.slack.com/archives/C02HLSDH3DZ/p1711019026080759

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,46 +21,11 @@ This App is meant to be a drop-in add-on that can be installed with default conf
...
-    - protocol: UDP
-      port: 53
-    - protocol: UDP
-      port: 1053
-```
-
 ## Known issues and limitations
 
-- The upstream application only works with `kube-proxy`-based clusters. It will NOT work when using other `Kubernetes Service` implementations such as `Cilium` for example.
+- The upstream application only works with `kube-proxy`-based clusters. It will NOT work when using other `Kubernetes Service` (e.g. `cilium`). This app works with `cilium` by using `CiliumLocalRedirectPolicy` enabled by default.
```
Comment (mcharriere) [inline README.md:26]: I'm not sure what information we are providing here. This is rather a configuration issue than a limitation.
In our case, with Cilium we followed this [guide](https://docs.cilium.io/en/stable/network/kubernetes/local-redirect-policy/#node-local-dns-cache). Other CNIs might need a different approach.
The upstream app is flexible enough to accommodate other mechanisms (it supports disabling the updates in iptables).

As I see it, non of the limitations described here are valid for our app (except the net-exporter one probably)

Comment (kopiczko) [reply README.md:26]: Delete this point?

Comment (ced0ps) [reply README.md:26]:
```suggestion
- The upstream application only works with `kube-proxy`-based clusters. This app has been made compatible with `cilium` as the cluster network proxy following [this guide](https://docs.cilium.io/en/stable/network/kubernetes/local-redirect-policy/#node-local-dns-cache).

Comment (ced0ps) [reply README.md:26]: I think it's worth noting how this app differs from upstream and how the functionality was achieved.

**Comment by kopiczko** [comment]
PTALA

**Review by mcharriere — APPROVED** [review]

**Pull request merged**